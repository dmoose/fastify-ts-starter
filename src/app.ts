import fastify from 'fastify'
import fastifySensible from 'fastify-sensible'
import config from './config'

const app = fastify({
  logger: config.get('env') !== 'test',
})

app.decorate('convict', config) // add the config to the server instance
app.register(fastifySensible)

export { app }
