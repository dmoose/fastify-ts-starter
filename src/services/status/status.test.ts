import { app } from '../../app'

import statusRoutes from './index'

describe('/status', () => {
  beforeAll(async () => {
    app.register(statusRoutes)
    await app.ready()
  })

  afterAll(async () => {
    app.close()
  })

  it('GET returns 200', async () => {
    const response = await app.inject({ method: 'GET', url: '/status' })
    expect(response.statusCode).toEqual(200)
  })

  it('POST returns 404', async () => {
    const response = await app.inject({ method: 'POST', url: '/status' })
    expect(response.statusCode).toEqual(404)
    expect(response.payload).toMatchSnapshot()
  })
})
