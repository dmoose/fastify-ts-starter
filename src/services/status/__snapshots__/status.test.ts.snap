// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/status POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/status not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
